### PR TITLE
Test for #74: Support fragments on Unions and Interfaces

### PR DESCRIPTION
**Pull request by danbruder**
Failing test case for https://github.com/Youshido/GraphQL/issues/74.

### CODE REVIEW

File: Tests/Library/Validator/ResolveValidatorTest.php [hunk 1]
```diff
@@ -80,6 +81,24 @@ public function testValidFragmentTypeWithComposite()
       $validator->assertValidFragmentForField($fragment, $fragmentReference, new NonNullType($userType));
     }
 
+    public function testValidFragmentTypeWithUnion()
+    {
+      $validator = new ResolveValidator(new ExecutionContext(new TestSchema()));
+
+      $unionType = new ObjectType([
```
Comment (fubhy) [inline Tests/Library/Validator/ResolveValidatorTest.php]: You named the variable $unionType but put an ObjectType in there. You should not be creating an Object Type at all. Instead, all we need is a union type by itself which we can then pass to the assertValidFragmentForField method.

File: Tests/Library/Validator/ResolveValidatorTest.php [hunk 2]
```diff
@@ -80,6 +81,24 @@ public function testValidFragmentTypeWithComposite()
...
+        'name' => 'Mammals',
+        'type'    => new TestUnionType(),
+        'resolve' => function ($value, $args, ResolveInfo $info) {
+          $testObject = new TestObjectType();
+          return [$testObject->getData()];
+        }
+      ]);
+
+      $fragment          = new Fragment('name', 'TestObject', ['name']);
```
Comment (fubhy) [inline Tests/Library/Validator/ResolveValidatorTest.php]: No need to pass a fields array (although it doesn't hurt I guess).

Comment (fubhy) [inline Tests/Library/Validator/ResolveValidatorTest.php]: No need to pass a fields array (although it doesn't hurt I guess).

**Comment by fubhy** [comment]
Great, thanks. Can you also add a union test that does the same for interfaces?

**Comment by viniychuk** [comment]
Please, run the test with the latest https://github.com/Youshido/GraphQL/releases/tag/v1.4, should be fixed.

**Comment by fubhy** [comment]
This works with unions now, but not yet with interfaces. Should be easier to fix now though. Will open a new PR. This can be closed imho.

**Pull request closed without merging**